pkg/backup/util/util.go

```diff
@@ -705,3 +695,92 @@ func isLogBackSupport(tikvImage string) bool {
 	}
 	return true
 }
+
+// GetContextForTerminationSignals get a context for some termination signals, and the context will become done after any of these signals triggered.
+func GetContextForTerminationSignals(op string) (context.Context, context.CancelFunc) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	sc := make(chan os.Signal, 1)
+	signal.Notify(sc,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+	go func() {
+		sig := <-sc
+		klog.Errorf("got signal %s to exit, %s will be canceled", sig, op)
+		cancel() // NOTE: the `Message` in `Status.Conditions` will contain `context canceled`.
+	}()
+	return ctx, cancel
+}
+
+// GetStorageRestorePath generate the path of a specific storage from Restore
+func GetStorageRestorePath(restore *v1alpha1.Restore) (string, error) {
```

Comment:
What is the difference of `GetStorageRestorePath` and `GetStorageBackupPath`? I think we can merge them to one function with parameter `StorageProvider`